src/transformers/generation/logits_process.py

```diff
@@ -1702,3 +1702,28 @@ def __call__(self, input_ids, scores):
         unconditional_logits = torch.nn.functional.log_softmax(logits[:, -1], dim=-1)
         out = self.guidance_scale * (scores - unconditional_logits) + unconditional_logits
         return out
+
+
+class BarkEarlyStoppingLogitsProcessor(LogitsProcessor):
+    r"""This processor will set every tokens' log probability other than the EOS token to `-inf`
+    when the probabiliy of the EOS token id is superior to min_eos_p.
+
+    Args:
+        generate_config (`GenerateConfig`):
+            The generate config used to generate the output. The following parameters are required:
+                eos_token_id (`int`, *optional*, defaults to 50257):
+                    The id of the *end-of-sequence* token.
+                min_eos_p (`float`, *optional*, defaults to None):
+                    Minimum end of speech threshold.
+    """
+
+    def __init__(self, generate_config):
+        self.eos_token_id = generate_config.eos_token_id
+        self.min_eos_p = generate_config.min_eos_p
+
+    @add_start_docstrings(LOGITS_PROCESSOR_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+        for k in range(input_ids.shape[0]):
+            if self.min_eos_p is not None and scores[k, self.eos_token_id] > self.min_eos_p:
+                scores[k, : self.eos_token_id] = -float("inf")
```

Comment:
`scores[k, self.eos_token_id] > self.min_eos_p` is probably wrong, since `scores` should be log probabilities. I would say that it should be `scores[k, self.eos_token_id]>log(min_eos_p)`.

@gante, can you confirm that?

Either way, you should probably ensure that `self.eos_token_id > 0` otherwise, their might be issue with the log.

Some other small remarks:
- You can directly return scores if `self.min_eos_p is None`, then you don't have to write `self.min_eos_p is not None` for each sample.
- `eos_token_id` is not necessarily the last id, i.e it is not necessarily equals to `scores.shape[1]`, so you also need to ensure other ids are set to `-inf`.
- You can make avoid the use of the loop here with something like this:

```python
if self.min_eos_p:
    # create scores full of -inf except for the eos_token_id
    early_stop_scores = torch.ones_like(scores)*-float("inf")
    early_stop_scores[:, self.eos_token_id] = scores[:, self.eos_token_id]

    do_early_stop = scores[:, self.eos_token_id, None] > threshold # todo: determine threshold

    scores = torch.where(do_early_stop, early_stop_scores, scores)

    return scores
```

Reply:
Implemented any for now.

Reply:
`eos_token_id` should never be a list of `int`, but only a `float`, since it is a threshold probability and should work with batches of any size. 
So for each sample of the batch, if `eos_token_id` probability of this sample is superior to the threshold, the non-EOS tokens scores of the sample should be `-inf`.

I believe that `scores = torch.where(do_early_stop, early_stop_scores, scores)` should do the job, but I might be wrong oc!

Reply:
Oh i see! Thanks, I misunderstood. Let me update that.